src/node/node_mut.rs

```diff
@@ -357,6 +357,285 @@ impl<'a, T> NodeMut<'a, T> {
         NodeRef::new(self.node_id, self.tree)
     }
 
+    /// Exchange positions with the next sibling.
+    ///
+    /// Returns true if swapped with a next sibling, returns false if this was
+    /// already the last sibling.
+    ///
+    /// ```
+    /// use slab_tree::tree::TreeBuilder;
+    ///
+    /// let mut tree = TreeBuilder::new().with_root(1).build();
+    /// let two_id = {
+    ///     let mut root = tree.root_mut().expect("root doesn't exist?");
+    ///     let two_id = root.append(2).node_id();
+    ///     root.append(3);
+    ///     root.append(4);
+    ///     two_id
+    /// };
+    /// assert_eq!(
+    ///     tree.root().unwrap().children().map(|child_ref| *child_ref.data())
+    ///         .collect::<Vec<i32>>(),
+    ///     vec![2, 3, 4]);
+    /// assert!(tree.get_mut(two_id).unwrap().swap_next_sibling());
+    /// assert_eq!(
+    ///     tree.root().unwrap().children().map(|child_ref| *child_ref.data())
+    ///         .collect::<Vec<i32>>(),
+    ///     vec![3, 2, 4]);
+    /// assert!(tree.get_mut(two_id).unwrap().swap_next_sibling());
+    /// assert_eq!(
+    ///   tree.root().unwrap().children().map(|child_ref| *child_ref.data())
+    ///         .collect::<Vec<i32>>(),
+    ///     vec![3, 4, 2]);
+    /// assert!(!tree.get_mut(two_id).unwrap().swap_next_sibling());
+    /// assert_eq!(
+    ///     tree.root().unwrap().children().map(|child_ref| *child_ref.data())
+    ///         .collect::<Vec<i32>>(),
+    ///     vec![3, 4, 2]);
```

Comment:
Do you mind adding assertions about the parent's first and last child (for this one and the other 3 new functions as well)?  I believe that's the only thing that you're not checking for here.  Otherwise this looks pretty great!